src/common/common/obd_config.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <iostream>
+#include <unordered_map>
+
+std::unordered_map<std::string, std::string> parseObdData(const std::string& obdData);
+std::string serializeObdData(const std::unordered_map<std::string, std::string>& obdData);
+
+class ObdConfig
+{
+		std::string m_filePath;
+	public:
+		ObdConfig(const std::string& filePath); // implicitly read config file
+		~ObdConfig(); // write back to file when the config goes out of scope
```

Comment:
Rule of 5 should apply here, the copy constructor/assignment operator and move constructor/ assignment operator should be also defined (as =delete) 

eg

`ObdConfig(const ObdConfig&) = delete;`

etc

https://en.cppreference.com/w/cpp/language/rule_of_three

Reply:
Is this necessary if all of the resources managed by `ObdConfig` are classes?